Wikipedia/Code/ArticleWebMessagingController.swift

```diff
@@ -86,26 +86,18 @@ class ArticleWebMessagingController: NSObject {
         webView?.evaluateJavaScript(js)
     }
     
-    func prepareForScroll(to anchor: String, highlight: Bool, completion: @escaping (Result<CGRect, Error>) -> Void) {
+    func prepareForScroll(to anchor: String, highlight: Bool, completion: @escaping (Result<Bool, Error>) -> Void) {
```

Comment:
This is just the callback for the initial JS execution, we don't get the actual value until the interaction handler event fires later. I initially tried `Result<Void, Error>` but the compiler wasn't happy - open to other suggestions here

Reply:
I found https://stackoverflow.com/a/57005869, it seems like .success(()) works. I handled it this exact same way in other areas so I pushed a commit fixing these.

Reply:
Awesome, thank you!